docs/proposals/network/README.md

```diff
@@ -0,0 +1,109 @@
+# Software Defined Network
+
+## Analysis
+
+### Examples
+
+- Flannel: good example, focused on networking
+- Calico
+- Weave
+- Cilium
+
+### Sources
+
+- [CNI](https://github.com/containernetworking/cni) also make a good framework for creating a new container networking project from scratch
+- [Networks - treeptik k8s](https://treeptik.gitbook.io/k8s/fundamentals/)
+- [Explain NS linux](https://www.youtube.com/watch?v=j_UUnlVC2Ss)
+- [Spec - CNI](https://www.cni.dev/docs/spec/)
+
+## Network domain
+
+### Glossary
+
+- *runtime* is the program responsible for executing CNI plugins.
+- *plugin* is a program that applies a specified network configuration.  
+  (it's a binary or an executable)
+
+### Definition
+
+The **orka SDN** implements the [`CNI`](https://www.cni.dev/docs/spec/) specification.
+
+It is a **CNI plugin** and must be called by e.g. the node Agent through the `CNI` specified
+command line interface. It configure workloads networking according to a set of `CNI`
+compliant configuration files, as described in the following sections.
+
+The goal is to be able to:
+
+- manage networks through the software
+- isolate instance/pods with isolated networks
+- apply network security policies
+
+The *plugins* must be installed on every node in the cluster.
+
+### Scope
+
+- Pod-to-Pod network packets on the same host
+	- Firewall (e.g. iptables)
+		- Need to ACCEPT FORWARDING traffic for pod CIDR because packets are dropped by the Linux kernel by default  
+		  (Linux treats network packets in non-default network namespaces as external packets by default)
+	- Allow outgoing internet
+- Cross-node pod-to-pod network packets
+- Port mapping / Selectors pods
+- DNS routing (e.g. ingress controller)
+- DNS resolution (e.g. coreDNS) -> pod-to-pod and pod-to-service
+	- "<NAME_SVC>.<NAMESPACE>.svc.cluster.local"
+	- "<POD_IP_WITH_DASH_INSTEAD_DOT>.<NAMESPACE>.pod.cluster.local"
+
+### Out of scope
+
+- RBAC, Network Policies
+- Monitoring network traffic (e.g. metrics-server)
+- Load balancing (e.g. service)
+- Load balancer (e.g. klipper, metallb)
+
+## API - Component interface
+
+### CNI
+
+#### Configuration
+
+Configuration files (`.conf`) are located in `/etc/cni/net.d/`.
+
+Example of a configuration file:
+
+```
+{
+    "cniVersion": "<VERSION>",
+    "name": "<CNI_NAME>",
+    "type": "<CNI_BINARY_NAME>",
+    // other variables to pass to plugin
```

Comment:
I guess this is the only thing left for you to clarify until we can merge that PR.